### PR TITLE
run ruff on all python files

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Instead of skipping large parts of our code, let's run ruff on everything.

This introduces some fixes that were easy enough and adds a series of `ruff.toml` files in various subdirectories with extra "ignores", depending on the needs of the code in those directories.

The eventual goal will be to remove those all and get all code up to the same set of standards, but this is a good start in that direction.

### CODE REVIEW

File: test/image-prepare [hunk 1]
```diff
@@ -137,21 +141,30 @@
         args += [
             # remove already installed packages
             "--run-command",
-            "if rpm -q cockpit-ws >/dev/null 2>&1; then rpm --erase --nodeps --verbose cockpit cockpit-ws cockpit-bridge cockpit-system; fi",
+            "if rpm -q cockpit-ws >/dev/null 2>&1; "
+            "then rpm --erase --nodeps --verbose cockpit cockpit-ws cockpit-bridge cockpit-system; "
+            "fi",
```
[the remark this answers is not in the document]
Comment (allisonkarlitskaya) [reply test/image-prepare:146]: To be honest, I kinda agree here.  I'll see if I can ignore this rule in ruff and maybe file an issue about it.

**Review by jelly — APPROVED** [review]

File: test/common/tap-cdp [hunk 2]
```diff
@@ -58,12 +56,6 @@ if opts.strip and opts.test.startswith(opts.strip):
 
 cdp = cdp.CDP("C.utf8")
 
-if sys.version[0] >= '3':
-    # with Python 3, don't rely on external locale, in case it is C
-    import codecs
-    sys.stdout = codecs.getwriter("utf-8")(sys.stdout.detach())
-    sys.stderr = codecs.getwriter("utf-8")(sys.stderr.detach())
-
```
[the remark this answers is not in the document]
Comment (allisonkarlitskaya) [reply test/common/tap-cdp:65]: Maybe?  `main` doesn't care about rhel7 anymore, though.

File: test/image-prepare (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinpitt) [reply test/image-prepare]: Oh yes, this is nicer, thanks! I'm also yearning for the day when we can drop all that rhel 8 "split build" hackery..

**Comment by martinpitt** [comment]
Meh, RHEL8 now fails to build:  `bash: line 2: -bs: command not found`

**Review by martinpitt — APPROVED** [review]
Noooice!

**Pull request merged**